Annotate some main-thread-only classes and protocols in WebKit API

geoffreygaren opened the pull request:
#### 13873b2f58fb7748c641b8f29ebc19e5ea73f267
<pre>
Annotate some main-thread-only classes and protocols in WebKit API
<a href="https://bugs.webkit.org/show_bug.cgi?id=273798">https://bugs.webkit.org/show_bug.cgi?id=273798</a>
<a href="https://rdar.apple.com/127629908">rdar://127629908</a>

Reviewed by Brady Eidson.

Most WebKit APIs are main-thread-only so let&apos;s say so for better
ergonomics and error checking for clients.

I manually tested these classes:

WKBackForwardList
WKBackForwardListItem
WKWebViewConfiguration
WKUserContentController
WKContentRuleList
WKContentRuleListStore
WKContentWorld
WKContextMenuElementInfo
WKDownload
WKFindConfiguration
WKFindResult
WKFrameInfo
WKHTTPCookieStore
WKNavigation
WKNavigationAction
WKNavigationResponse
WKPreferences
WKProcessPool
WKScriptMessage
WKWebsiteDataStore
WKOpenPanelParameters
WKPDFConfiguration
WKSecurityOrigin
WKSnapshotConfiguration
WKUserScript
WKWebpagePreferences
WKWebsiteDataRecord
WKWebView
WKWindowFeatures

...and these callback-to-WebKit APIs:

WKDownloadDelegate
   - decideDestinationUsingResponse
   - willPerformHTTPRedirection
   - didReceiveAuthenticationChallenge

WKUIDelegate
   - runJavaScriptAlertPanelWithMessage
   - runJavaScriptConfirmPanelWithMessage
   - runJavaScriptTextInputPanelWithPrompt
   - requestMediaCapturePermissionForOrigin
   - requestDeviceOrientationAndMotionPermissionForOrigin
   - contextMenuConfigurationForElement
   - showLockdownModeFirstUseMessage
   - runOpenPanelWithParameters

WKScriptMessageHandlerWithReply
   - didReceiveScriptMessage

...and verified that they compile when used in &quot;isolated&quot; (i.e. main thread)
code and fail to compile when used in &quot;nonisolated&quot; (i.e. secondary thread)
code.

I manually tested these protocols:

WKHTTPCookieStoreObserver
WKUIDelegate
WKURLSchemeHandler
WKScriptMessageHandler
WKScriptMessageHandlerWithReply
WKDownloadDelegate
WKNavigationDelegate

...and these callback-to-client APIs:

WKContentRuleListStore
   - compileContentRuleListForIdentifier
   - lookUpContentRuleListForIdentifier
   - removeContentRuleListForIdentifier
   - getAvailableContentRuleListIdentifiers

WKDownload
   - cancel

WKHTTPCookieStore
   - getAllCookies
   - setCookie
   - deleteCookie
   - setCookiePolicy
   - getCookiePolicy

WKWebsiteDataStore
   - fetchDataRecordsOfTypes
   - removeDataOfTypes
   - removeDataOfTypes
   - removeDataStoreForIdentifier
   - fetchAllDataStoreIdentifiers

* WKWebView.h
   - evaluateJavaScript
   - evaluateJavaScript
   - callAsyncJavaScript
   - closeAllMediaPresentationsWithCompletionHandler
   - pauseAllMediaPlaybackWithCompletionHandler
   - setAllMediaPlaybackSuspended
   - requestMediaPlaybackStateWithCompletionHandler
   - setCameraCaptureState
   - setMicrophoneCaptureState
   - takeSnapshotWithConfiguration
   - createPDFWithConfiguration
   - createWebArchiveDataWithCompletionHandler
   - findString
   - startDownloadUsingRequest
   - resumeDownloadFromResumeData

...and verified that they compile when implemented in &quot;isolated&quot; (i.e. main
thread) code.

These classes are thread safe and need no annotation:

WKURLSchemeTask

Canonical link: <a href="https://commits.webkit.org/278930@main">https://commits.webkit.org/278930@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3258b4491521277a47d858fdd314a5e549efd46d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51363 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30669 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3707 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54626 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/2053 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36984 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1732 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41841 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/1226 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53462 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28314 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/44291 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22952 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25633 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/246 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/47608 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1626 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56216 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26480 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/49238 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27721 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44356 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/48416 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11376 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28612 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27456 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/Shared/API/Cocoa/WKFoundation.h:
```diff
@@ -56,6 +56,12 @@
 #endif
 
+#ifdef NS_SWIFT_UI_ACTOR
+#define WK_SWIFT_UI_ACTOR NS_SWIFT_UI_ACTOR
```
nice!

rr-codes commented on Source/WebKit/UIProcess/API/Cocoa/WKDownload.h:
```diff
@@ -34,6 +34,7 @@
 NS_ASSUME_NONNULL_BEGIN
 
 WK_CLASS_AVAILABLE(macos(11.3), ios(14.5))
+WK_SWIFT_UI_ACTOR
```
does this have to go through API review?

geoffreygaren replied:
Yes.

geoffreygaren replied:
API review is ✅.

geoffreygaren committed:
Annotate some main-thread-only classes and protocols in WebKit API
https://bugs.webkit.org/show_bug.cgi?id=273798
rdar://127629908

Reviewed by Brady Eidson.

Most WebKit APIs are main-thread-only so let's say so for better
ergonomics and error checking for clients.

I manually tested these classes:

WKBackForwardList
WKBackForwardListItem
WKWebViewConfiguration
WKUserContentController
WKContentRuleList
WKContentRuleListStore
WKContentWorld
WKContextMenuElementInfo
WKDownload
WKFindConfiguration
WKFindResult
WKFrameInfo
WKHTTPCookieStore
WKNavigation
WKNavigationAction
WKNavigationResponse
WKPreferences
WKProcessPool
WKScriptMessage
WKWebsiteDataStore
WKOpenPanelParameters
WKPDFConfiguration
WKSecurityOrigin
WKSnapshotConfiguration
WKUserScript
WKWebpagePreferences
WKWebsiteDataRecord
WKWebView
WKWindowFeatures

...and these callback-to-WebKit APIs:

WKDownloadDelegate
   - decideDestinationUsingResponse
   - willPerformHTTPRedirection
   - didReceiveAuthenticationChallenge

WKUIDelegate
   - runJavaScriptAlertPanelWithMessage
   - runJavaScriptConfirmPanelWithMessage
   - runJavaScriptTextInputPanelWithPrompt
   - requestMediaCapturePermissionForOrigin
   - requestDeviceOrientationAndMotionPermissionForOrigin
   - contextMenuConfigurationForElement
   - showLockdownModeFirstUseMessage
   - runOpenPanelWithParameters

WKScriptMessageHandlerWithReply
   - didReceiveScriptMessage

...and verified that they compile when used in "isolated" (i.e. main thread)
code and fail to compile when used in "nonisolated" (i.e. secondary thread)
code.

I manually tested these protocols:

WKHTTPCookieStoreObserver
WKUIDelegate
WKURLSchemeHandler
WKScriptMessageHandler
WKScriptMessageHandlerWithReply
WKDownloadDelegate
WKNavigationDelegate

...and these callback-to-client APIs:

WKContentRuleListStore
   - compileContentRuleListForIdentifier
   - lookUpContentRuleListForIdentifier
   - removeContentRuleListForIdentifier
   - getAvailableContentRuleListIdentifiers

WKDownload
   - cancel

WKHTTPCookieStore
   - getAllCookies
   - setCookie
   - deleteCookie
   - setCookiePolicy
   - getCookiePolicy

WKWebsiteDataStore
   - fetchDataRecordsOfTypes
   - removeDataOfTypes
   - removeDataOfTypes
   - removeDataStoreForIdentifier
   - fetchAllDataStoreIdentifiers

* WKWebView.h
   - evaluateJavaScript
   - evaluateJavaScript
   - callAsyncJavaScript
   - closeAllMediaPresentationsWithCompletionHandler
   - pauseAllMediaPlaybackWithCompletionHandler
   - setAllMediaPlaybackSuspended
   - requestMediaPlaybackStateWithCompletionHandler
   - setCameraCaptureState
   - setMicrophoneCaptureState
   - takeSnapshotWithConfiguration
   - createPDFWithConfiguration
   - createWebArchiveDataWithCompletionHandler
   - findString
   - startDownloadUsingRequest
   - resumeDownloadFromResumeData

...and verified that they compile when implemented in "isolated" (i.e. main
thread) code.

These classes are thread safe and need no annotation:

WKURLSchemeTask

Canonical link: https://commits.webkit.org/278930@main
